p2pserver/net/netserver/netserver.go

```diff
@@ -483,9 +483,9 @@ func (this *NetServer) RemoveFromConnectingList(addr string) {
 	this.ConnectingNodes.Lock()
 	defer this.ConnectingNodes.Unlock()
 	addrs := []string{}
-	for i, a := range this.ConnectingAddrs {
-		if strings.Compare(a, addr) == 0 {
-			addrs = append(this.ConnectingAddrs[:i], this.ConnectingAddrs[i+1:]...)
+	for _, a := range this.ConnectingAddrs {
+		if strings.Compare(a, addr) != 0 {
+			addrs = append(addrs, a)
```

Comment:
Why not use this.ConnectingAddrs = append(this.ConnectingAddrs[:i], this.ConnectingAddrs[i+1:]...), instead of using temp variable addrs ?

Reply:
Because if you change the this.ConnectingAddrs which is used for loop, it is dangerous. And in this case, it also can not remove duplicated elements.

Reply:
Ok. So please use addrs := this.ConnectingAddrs[:0] to avoid allocating a new underlying array

Reply:
btw: you can use ` if a != addr` directly

Reply:
That's right, thanks all